src/plugins/collision/narrow_phase.rs

```diff
@@ -139,6 +140,40 @@ pub fn collect_collisions(
             .into_iter()
             .flatten();
         collisions.extend(new_collisions);
+
+        let existing_unmoving_collisions = collisions
+            .0
+            .keys()
+            .map(|(e1, e2)| {
+                if let Ok([bundle1, bundle2]) = bodies.get_many([*e1, *e2]) {
+                    if bundle1.0.is_changed()
+                        || bundle1.2.is_changed()
+                        || bundle2.0.is_changed()
+                        || bundle2.2.is_changed()
+                    {
+                        None
+                    } else {
+                        // if the position and rotation have not changed for both bundles, renew the collision
+                        let previous_contact = collisions.get_internal().get(&(*e1, *e2));
+                        Some(Contacts {
+                            entity1: *e1,
+                            entity2: *e2,
+                            during_current_frame: true,
+                            during_current_substep: true,
+                            during_previous_frame: previous_contact
+                                .map_or(false, |c| c.during_previous_frame),
+                            manifolds: vec![],
+                        })
+                    }
+                } else {
+                    None
+                }
+            })
+            .filter(Option::is_some)
+            .map(Option::unwrap)
+            .collect_vec();
+
+        collisions.extend(existing_unmoving_collisions);
```

Comment:
This _shouldn't_ introduce duplicate collisions since `broad_collision_pairs` doesn't include pairs of inactive entities, where "inactive" is determined by the same `is_changed` checks

https://github.com/Jondolf/bevy_xpbd/blob/088facfe3761235f0b297bc1493927a12a6fe21b/src/plugins/collision/broad_phase.rs#L192

https://github.com/Jondolf/bevy_xpbd/blob/088facfe3761235f0b297bc1493927a12a6fe21b/src/plugins/collision/broad_phase.rs#L259-L262